### PR TITLE
Update Identity tests for EF breaking changes

### PR DESCRIPTION
**Pull request by ajcvickers**
Specifically:
* Removal of Relational()
* Query changes

There is some kind of flakiness still impacting these tests. It may be a race condition in EF query generation, or it may be an issue with the test infrastructure

Also, one database error page test is currently disabled.

### CODE REVIEW

**Review by smitpatel — APPROVED** [review]

File: src/Identity/EntityFrameworkCore/src/UserOnlyStore.cs [hunk 1]
```diff
@@ -311,7 +312,8 @@ public async override Task<IList<Claim>> GetClaimsAsync(TUser user, Cancellation
                 throw new ArgumentNullException(nameof(user));
             }
 
-            return await UserClaims.Where(uc => uc.UserId.Equals(user.Id)).Select(c => c.ToClaim()).ToListAsync(cancellationToken);
+            // TODO: Fix once EF query works
```
Comment (Tratcher) [inline src/Identity/EntityFrameworkCore/src/UserOnlyStore.cs]: Include an issue link with every TODO

Comment (ajcvickers) [reply src/Identity/EntityFrameworkCore/src/UserOnlyStore.cs]: Issues don't exist yet, and it's more urgent to get this in. I will make sure there are issues.

**Review by HaoK — APPROVED** [review]
I'm assuming most of these will be addressed by EF in preview7?

**Comment by ajcvickers** [comment]
@HaoK Yes, or before.

**Comment by ajcvickers** [comment]
@HaoK @natemcmaster I can't figure out how to get the ref assemblies failure to pass. I can't see any differences, so I don't understand why it is failing. Any ideas?

**Comment by HaoK** [comment]
Try running `dotnet msbuild /t:GenerateReferenceSource` from the src directories of the refs you want to update, if that works, it usually does the trick

**Comment by HaoK** [comment]
Okay I see the issue, you made something not async, so you need to remove the 

```
[System.Diagnostics.DebuggerStepThroughAttribute]
```

**Comment by HaoK** [comment]
Remove  `[System.Diagnostics.DebuggerStepThroughAttribute]` from
```
         public override System.Threading.Tasks.Task<System.Collections.Generic.IList<System.Security.Claims.Claim>> GetClaimsAsync(TUser user, System.Threading.CancellationToken cancellationToken = default(System.Threading.CancellationToken)) { throw null; }
``` 
in src/Identity/EntityFrameworkCore/ref/Microsoft.AspNetCore.Identity.EntityFrameworkCore.netcoreapp3.0.cs.

**Comment by ajcvickers** [comment]
@natemcmaster @dougbu @JunTaoLuo Now this is merged we should be good to take the latest EF dependency here.

/cc @Pilchie @smitpatel @divega

**Comment by natemcmaster** [comment]
Thanks!

**Comment by dougbu** [comment]
I created #10682 as a draft PR to see how things go.

@ajcvickers EF Core is behind on its CoreFx, core-setup and Extensions dependencies. Any objection to EveryBuild there? (We left it and EF6 when we sped everything else up.)

**Comment by smitpatel** [comment]
Every build is too much noise.

**Comment by ajcvickers** [comment]
@dougbu Yes, every build is fine while we are working towards coherence for the preview.

@smitpatel We will switch it off once we get to where we need to be for preview.

**Comment by dougbu** [comment]
Done

**Pull request merged**